### ISSUE TITLE
Render layers in 2D are probably ignored 

### ISSUE THREAD

**Issue by Pieresqi** [james-j-obrien/bevy_vector_shapes#15]
Hello,

Let's take this code:

```rust
use bevy::{
    core_pipeline::clear_color::ClearColorConfig,
    prelude::*,
    render::{camera::RenderTarget, texture::ImageSampler, view::RenderLayers},
};
use bevy_vector_shapes::{
    prelude::{Canvas, ShapePainter},
    shapes::RectPainter,
    ShapePlugin,
};

fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .add_plugin(ShapePlugin::default())
        .insert_resource(ClearColor(Color::BLACK))
        //.add_system(setup_2d.on_startup())
        //.add_system(draw_shapes_2d)
        //.add_system(setup_3d.on_startup())
        //.add_system(draw_shapes_3d)
        .run();
}

fn setup_2d(mut commands: Commands, mut images: ResMut<Assets<Image>>) {
    let image_handle =
        Canvas::create_image(images.as_mut(), 512, 512, ImageSampler::Default, false);

    commands.spawn((
        Camera2dBundle {
            camera_2d: Camera2d {
                clear_color: ClearColorConfig::Custom(Color::WHITE),
                ..default()
            },
            camera: Camera {
                order: -1,
                target: RenderTarget::Image(image_handle.clone()),
                ..default()
            },
            ..default()
        },
        RenderLayers::layer(1),
    ));

    commands.spawn((
        SpriteBundle {
            texture: image_handle,
            transform: Transform::from_xyz(500.0, 0.0, 0.0),
            ..default()
        },
    ));

    commands.spawn(Camera2dBundle::default());
}

fn setup_3d(
    mut commands: Commands,
    mut meshes: ResMut<Assets<Mesh>>,
    mut materials: ResMut<Assets<StandardMaterial>>,
    mut images: ResMut<Assets<Image>>,
) {
    let image_handle =
        Canvas::create_image(images.as_mut(), 512, 512, ImageSampler::Default, false);

    commands.spawn((
        Camera3dBundle {
            camera_3d: Camera3d {
                clear_color: ClearColorConfig::Custom(Color::WHITE),
                ..default()
            },
            camera: Camera {
                order: -1,
                target: RenderTarget::Image(image_handle.clone()),
                ..default()
            },
            ..default()
        },
        RenderLayers::layer(1),
    ));

    let cube_size = 4.0;
    let cube_handle = meshes.add(Mesh::from(shape::Box::new(cube_size, cube_size, cube_size)));

    let material_handle = materials.add(StandardMaterial {
        base_color_texture: Some(image_handle),
        unlit: true,
        ..default()
    });

    commands.spawn((
        PbrBundle {
            mesh: cube_handle,
            material: material_handle,
            transform: Transform::from_xyz(5.0, 0.0, 0.0),
            ..default()
        },
    ));

    commands.spawn(Camera3dBundle {
        transform: Transform::from_xyz(0.0, 0.0, 15.0),
        ..default()
    });
}

fn draw_shapes_2d(mut painter: ShapePainter) {
    painter.reset();
    painter.set_2d();
    painter.render_layers = Some(RenderLayers::layer(1));
    painter.color = Color::RED;
    painter.rect(Vec2 { x: 100.0, y: 100.0 });
}

fn draw_shapes_3d(mut painter: ShapePainter) {
    painter.reset();
    painter.set_3d();
    painter.render_layers = Some(RenderLayers::layer(1));
    painter.color = Color::RED;
    // Move the painter so it's not inside the camera
    painter.set_translation(Vec3::NEG_Z);
    painter.rect(Vec2 { x: 0.1, y: 0.1 });
}
```

It has 2 pairs of systems, 1st for 2d and 2nd for 3d.

3D systems work fine. Painter painted on cube.
2D systems not so much. Painter painted on square but also on "main" camera.

I think painter ignores render layers in 2D, or it's some other bug. Or I have it set up wrong...

**Comment by james-j-obrien** [comment]
You are completely correct, there is a fix I made in the 3d render pipeline that I did not mirror correctly to the 3d pipeline. I pushed a fix to main, let me know if that fixes it for you. If you haven't used git dependecies before it should look something like this in your Cargo.toml:
```toml
bevy_vector_shapes = { git = "https://github.com/james-j-obrien/bevy_vector_shapes.git" }
```
Thanks for the example code it made this very quick to debug and test.
